.github/scripts/sync_distributed_folder_prototype.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+ORIG_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+git checkout fbcode/pytorch-stable-prototype
+for SHA in $(git log 4333e122d4b74cdf84351ed2907045c6a767b4cd..main --pretty="%h" --reverse -- torch/distributed torch/csrc/distributed test/distributed test/cpp/c10d benchmarks/distributed)
+do
+    if git merge-base --is-ancestor $SHA HEAD || [[ $(git log --grep="(cherry picked from commit $SHA") ]]
```

Comment:
im not familiar enough with git merge-base, what is the logic here?

Reply:
i think its saying if `SHA` is already present on the stable branch, don't cherry pick it?

Reply:
Yeah that's basically it, I'll add a comment